src/Renci.SshNet/Abstractions/SocketAbstraction.cs

```diff
@@ -59,7 +59,31 @@ public static async Task ConnectAsync(Socket socket, IPEndPoint remoteEndpoint,
 
         private static void ConnectCore(Socket socket, IPEndPoint remoteEndpoint, TimeSpan connectTimeout, bool ownsSocket)
         {
-#if FEATURE_SOCKET_EAP
+#if FEATURE_SOCKET_TAP
+            AsyncHelper.RunSync(async () =>
+            {
+                using var ctsTs = new CancellationTokenSource(connectTimeout);
+
+                try
+                {
+                    await socket.ConnectAsync(remoteEndpoint, ctsTs.Token).ConfigureAwait(false);
+                }
+                catch (OperationCanceledException)
+                {
+                    throw new SshOperationTimeoutException(
+                        string.Format(CultureInfo.InvariantCulture,
+                                      "Connection failed to establish within {0:F0} milliseconds.",
+                                      connectTimeout.TotalMilliseconds));
+                }
+                finally
+                {
+                    if (!socket.Connected && ownsSocket)
+                    {
+                        socket.Dispose();
+                    }
+                }
+            });
+#elif FEATURE_SOCKET_EAP
             var connectCompleted = new ManualResetEvent(initialState: false);
             var args = new SocketAsyncEventArgs
```

Comment:
`FEATURE_SOCKET_EAP` uses `socket.ConnectAsync(args)` below, but this is not an official implementation, even for .NET 7. We have a custom extension for this in `SocketExtensions`. I think that we can improve this for modern .NETs. I will propose a better solution.

Reply:
`SocketAsyncEventArgs` is an official implementation, I think `ConnectCore` under `FEATURE_SOCKET_EAP` is fine even in modern .NET

`SocketExtensions.AwaitableSocketAsyncEventArgs` is a custom implementation but it is unnecessary in its current usage (the idea is to cache these objects but we create a new one each time). Indeed we could improve it.